javaCarAPI/src/test/java/apitest/CarAPITest.java

```diff
@@ -0,0 +1,141 @@
+package apitest;
+
+import api.CarAPI;
+import org.eclipse.paho.client.mqttv3.MqttException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.opentest4j.AssertionFailedError;
+import api.sensor.*;
+
+import java.util.Random;
+import java.util.concurrent.CountDownLatch;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+public class CarAPITest {
+
+    private final String CAR_NAME = "car";
+    private final String SERVER = "127.0.0.1";
+    private final int PORT = 1883;
+
+    private MockCar mockCar;
+    private CarAPI carAPI;
+
+    @BeforeEach
+    public void initialise() throws MqttException {
+
+        mockCar = new MockCar(SERVER,PORT,CAR_NAME);
+        carAPI = new CarAPI(SERVER,CAR_NAME);
+
+    }
+
+
+    @Test
+    public void testGyroscopeListener() throws Exception{
+
+        CountDownLatch latch = new CountDownLatch(1);
+        final double testValue = randomDouble();
+        final TestResult result = new TestResult();
+        carAPI.addGyroscopeListener(d -> {
```

Comment:
Consider renaming the lambda variables to something more meaningful to make the functions easier to read.